src/common/market/order.js

```diff
@@ -841,12 +845,104 @@ const getMatchableVolume = async (
   }
 };
 
+export const estimateMatchOrders = async (
+  contracts,
+  voucherHubAddress,
+  appOrder,
+  datasetOrder = NULL_DATASETORDER,
+  workerpoolOrder,
+  requestOrder,
+  useVoucher,
+) => {
+  const [vAppOrder, vDatasetOrder, vWorkerpoolOrder, vRequestOrder] =
+    await Promise.all([
+      signedApporderSchema().validate(appOrder),
+      signedDatasetorderSchema().validate(datasetOrder),
+      signedWorkerpoolorderSchema().validate(workerpoolOrder),
+      signedRequestorderSchema().validate(requestOrder),
+    ]);
+  const matchableVolume = await getMatchableVolume(
+    contracts,
+    vAppOrder,
+    vDatasetOrder,
+    vWorkerpoolOrder,
+    vRequestOrder,
+  );
+  const totalCost = getMatchOrdersTotalCost(
+    matchableVolume,
+    vAppOrder.appprice,
+    vDatasetOrder.datasetprice,
+    vWorkerpoolOrder.workerpoolprice,
+  );
+  let sponsoredCost = new BN(0);
+  if (useVoucher) {
+    const voucherAddress = await fetchVoucherAddress(
+      contracts,
+      voucherHubAddress,
+      requestOrder.requester,
+    );
+    if (!voucherAddress) {
+      return { total: totalCost, sponsored: sponsoredCost };
+    }
+    const voucherContract = await getVoucherContract(contracts, voucherAddress);
+    const voucherExpiration = voucherContract.getExpiration();
+    const now = Math.floor(Date.now() / 1000);
+    if (voucherExpiration <= now) {
+      return { total: totalCost, sponsored: sponsoredCost };
+    }
+    const [balance, voucherTypeId] = await Promise.all([
+      voucherContract.getBalance(),
+      voucherContract.getType(),
+    ]);
+    const voucherHubContract = getVoucherHubContract(
+      contracts,
+      voucherHubAddress,
+    );
+
+    const [
+      isAppEligibleToMatchOrdersSponsoring,
+      isDatasetEligibleToMatchOrdersSponsoring,
+      isWorkerpoolEligibleToMatchOrdersSponsoring,
+    ] = await Promise.all([
+      voucherHubContract.isAssetEligibleToMatchOrdersSponsoring(
+        voucherTypeId,
+        appOrder.app,
+      ),
+
+      voucherHubContract.isAssetEligibleToMatchOrdersSponsoring(
+        voucherTypeId,
+        datasetOrder.dataset,
+      ),
+
+      voucherHubContract.isAssetEligibleToMatchOrdersSponsoring(
+        voucherTypeId,
+        workerpoolOrder.workerpool,
+      ),
+    ]);
+
+    sponsoredCost = getMatchOrdersSponsoredCostWithVoucher(
+      vAppOrder,
+      vDatasetOrder,
+      vWorkerpoolOrder,
+      matchableVolume,
+      balance,
+      isAppEligibleToMatchOrdersSponsoring,
+      isDatasetEligibleToMatchOrdersSponsoring,
+      isWorkerpoolEligibleToMatchOrdersSponsoring,
+    );
+  }
+
+  return { total: totalCost, sponsored: sponsoredCost };
+};
+
```

Comment:
This change is due to moving the previously implemented estimateMatchOrder function to be used in matchOrder. the removed part is visible at the bottom of the changes.